File: week6/objective-c/PulseSecureVpnActionOperation.m
- (void)main {
    // Registers appHasConnectedToPulseSecureVpn as a handler for kPulseSecureVpnConnectedNotification
    [[NSNotificationCenter defaultCenter] addObserver:self
                                             selector:@selector(appHasConnectedToPulseSecureVpn)
                                                 name:kPulseSecureVpnConnectedNotification
                                               object:nil];
    
    self.pulseSecureVpnManager = [[PulseSecureVpnManager alloc] init];
    [self.pulseSecureVpnManager connectProxy];
}

- (void) appHasConnectedToPulseSecureVpn {
    // Done!
    [self finish];
    return;
}

File: week6/javascript/PulseSecureVpnActionOperation.js
function main() {
    var appHasConnectedToPulseSecureVpn = function() {
        // Done!
        return;
    }

    // Registers appHasConnectedToPulseSecureVpn as a handler for kPulseSecureVpnConnectedNotification
    var pulseSecureVpnManager = new PulseSecureVpnManager(appHasConnectedToPulseSecureVpn);
    pulseSecureVpnManager.connectProxy();
}